rs/nervous_system/agent/src/agent_impl.rs

```diff
@@ -139,4 +139,17 @@ impl CallCanisters for Agent {
     fn caller(&self) -> Result<Principal, Self::Error> {
         self.get_principal().map_err(Self::Error::Identity)
     }
+
+    async fn check_canister_exists(
+        &self,
+        canister_id: impl Into<Principal> + Send,
+    ) -> Result<bool, Self::Error> {
+        self.read_state_canister_info(canister_id.into(), "module_hash")
```

Comment:
Is this the correct way to check that a canister with the given ID exists?

Reply:
This checks that the canister not only exists, but also has some code installed into it. 

If you just needed to know that is was created, you could check its controllers.

Reply:
I guess checking that the code was installed is not a bad thing too

Reply:
Initially, I was considering using the `canister_info` method, but turned out it panics instead of returning `Err`. Perhaps, the more proper way to approach this method to panic in the first place

Reply:
Do you mean the PocketIC canister_info method, or something else?

Reply:
Yes, `canister_info` from `ic-nervous-agent` which calls PocketIC's `canister_info` internally

Reply:
Thanks for clarifying. No actions required

FYI, since PocketIC is defined in the same repo, we are able to make meaningful changes to it, if necessary.